Tests/LibC/TestScanf.cpp

```diff
@@ -251,3 +269,63 @@ TEST_CASE(scanf)
     for (auto& test : test_suites)
         do_one_test(test);
 }
+
+TEST_CASE(scanf_char)
+{
+    // %c should not put a null terminator
+    char buf[5] = "xxxx";
+    sscanf("abc", "%c", buf);
+    EXPECT_EQ(strcmp(buf, "axxx"), 0);
+    sscanf(" a bc", "%3c", buf);
+    EXPECT_EQ(strcmp(buf, " a x"), 0);
+}
+
+struct TestSuiteFloat {
+    char const* format;
+    char const* input;
+    double expected;
+};
+
+TestSuiteFloat const test_suites_float[] {
+    { "%f", "42.3456", 42.3456 },
+    { "%lf", "42.3456", 42.3456 },
+    { "%Lf", "42.3456", 42.3456 },
+    // Width specifiers
+    { "%3f", "4.234", 4.2 },
+    { "%3f", "  4.234", 4.2 },
+};
+
+static void expect_double_eq(double expected, double actual)
```

Comment:
we have `EXPECT_APPROXIMATE`, is that usable here?

Reply:
Thanks for the review!
I replaced `expect_double_eq` with `EXPECT_APPROXIMATE`. It works with clang on my Ubuntu machine.